### ISSUE TITLE
Snapshot updates wrongly trigger a statefulset update 

### ISSUE THREAD

**Issue by sebbonnet** [sky-uk/cassandra-operator#134]
Adding, removing or updating Snapshot properties to a `cassandra` cluster definition triggers a cluster update operation even though no other cluster properties have changed. This results in patching the statefulset, which has no effect as far as the cluster is concerned as nothing is changed in the statefulet.

However this has 2 side-effects from the cassandra-operator point of view:
- it creates an unnecessary `WaitingForStatefulSetChange` event
- it waits `max(*readinessProbe.PeriodSeconds, 5)`, meaning the next cluster event is delayed by this much

We should change the behaviour so the above 2 operations do not take place.

**Comment by sebbonnet** [comment]
Picking this up as this make some tests fail when run against an integrated environment where the readiness probe timeout is 30s. Tests in `snapshot_test` suite that modify Snapshot fail as they do not expect an additional 30s wait before the snapshot change event is fired

**Comment by sebbonnet** [comment]
Turns out this is actually due to the way we apply defaults to `cassandra` definition.
In case where both liveness and readiness probes are `nil`, defaults are being applied as the `update cluster` operation is being executed which results in applying the defaults to the `newCluster`: see https://github.com/sky-uk/cassandra-operator/blob/v0.70.1-alpha.2/cassandra-operator/pkg/operator/operations/update_cluster.go#L28
So I think it would make sense to play #133 before.

While investigating I also spotted that updates to our sidecar image would discarded.
We can also tackled the issue as part of this
